src/benchmark/tpcds_benchmark.cpp

```diff
@@ -0,0 +1,172 @@
+  Assert(std::filesystem::exists(std::filesystem::path{table_path + "/call_center.csv.json"}),
+         "Table schemes have to be available.");
+
+  if (!data_files_available(table_path)) {
+    if (std::filesystem::exists(std::filesystem::path{binary_directory + "/dsdgen"})) {
+      const auto files_setup_return =
+          system(("cd " + binary_directory + " && ./dsdgen -scale " + std::to_string(scale_factor) +
+                  " -dir ../resources/benchmark/tpcds/tables -terminate n -verbose -suffix .csv -f")
+                     .c_str());
+      Assert(files_setup_return == 0, "Generating table data files failed.");
+    } else {
+      Fail("Could not find 'dsdgen' in your build directory. Did you run the benchmark from the project root dir?");
+    }
+  }
+
+  Assert(data_files_available(table_path), "Generating table data files failed.");
+
+  auto query_generator = std::make_unique<FileBasedBenchmarkItemRunner>(config, query_path, filename_blacklist());
+  auto table_generator = std::make_unique<FileBasedTableGenerator>(config, table_path);
+  auto benchmark_runner = BenchmarkRunner{*config, std::move(query_generator), std::move(table_generator), context};
+
+  if (config->verify) {
+    add_indices_to_sqlite("resources/benchmark/tpcds/schema.sql", "resources/benchmark/tpcds/create_indices.sql",
+                          benchmark_runner);
+  }
+
+  std::cout << "done." << std::endl;
+
+  benchmark_runner.run();
+}
+
+namespace {
+
+bool data_files_available(const std::string& table_path) {
+  for (const auto& table : {"call_center",
+                            "catalog_page",
+                            "catalog_returns",
+                            "catalog_sales",
+                            "customer_address",
+                            "customer",
+                            "customer_demographics",
+                            "date_dim",
+                            "household_demographics",
+                            "income_band",
+                            "inventory",
+                            "item",
+                            "promotion",
+                            "reason",
+                            "ship_mode",
+                            "store",
+                            "store_returns",
+                            "store_sales",
+                            "time_dim",
+                            "warehouse",
+                            "web_page",
+                            "web_returns",
+                            "web_sales",
+                            "web_site"}) {
+    if (!std::filesystem::exists(table_path + "/" + table + ".csv")) {
+      return false;
+    }
+  }
+  return true;
+}
+
+const std::unordered_set<std::string> filename_blacklist() {
+  auto filename_blacklist = std::unordered_set<std::string>{};
+  const auto blacklist_file_path = "resources/benchmark/tpcds/query_blacklist.cfg";
```

Comment:
assuming you wrote this query_blacklist.cfg yourself to exclude queries hyrise doesn't support (add a comment what filename_blacklist() does?) - we should use JSON instead of inventing our own "fileformat" (I see comment support below).

Reply:
I find the current format more readable than `[{query: 15, executable: false}, ...]`. The "comment support" is for excluding queries (similar to the sqlparser tests). Not sure what we could gain from using JSON (and if it's worth spending time on it).